ms-common/src/main/java/net/trajano/ms/common/beans/DefaultAssertionRequiredFunction.java

```diff
@@ -16,8 +25,30 @@
     @Override
     public boolean test(final ResourceInfo resourceInfo) {
 
-        return resourceInfo.getResourceMethod().getAnnotation(JwtNotRequired.class) == null &&
-            resourceInfo.getResourceClass().getAnnotation(JwtNotRequired.class) == null;
+        final Method resourceMethod = resourceInfo.getResourceMethod();
+        final Class<?> resourceClass = resourceInfo.getResourceClass();
+
+        final boolean resourceMethodHasRolesAllowed = resourceMethod.getAnnotation(RolesAllowed.class) != null;
+        final boolean resourceClassHasRolesAllowed = resourceClass.getAnnotation(RolesAllowed.class) != null;
+
+        final boolean resourceMethodHasPermitAll = resourceMethod.getAnnotation(PermitAll.class) != null;
+        final boolean resourceClassHasPermitAll = resourceClass.getAnnotation(PermitAll.class) != null;
+
+        if (resourceMethodHasRolesAllowed && resourceMethodHasPermitAll) {
+            throw new IllegalArgumentException("The resource method " + resourceMethod + " may not have both @RolesAllowed and @PermitAll annotations.");
+        } else if (resourceClassHasRolesAllowed && resourceClassHasPermitAll) {
+            throw new IllegalArgumentException("The resource class " + resourceClass + " may not have both @RolesAllowed and @PermitAll annotations.");
+        } else {
+            // resourceMethodHasRolesAllowed OR
+            // resourceClassHasRolesAllowed && !resourceMethodHasPermitAll OR
+            // !resourceMethodHasRolesAllowed && !resourceClassHasRolesAllowed && !resourceMethodHasPermitAll && !resourceClassHasPermitAll
+
+            // a || ( b && !c ) || ( !a && !b && !c && !d)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)